diff --git a/dist/changesets-cli.cjs.js b/dist/changesets-cli.cjs.js
index 02083f394e874d25ca5ec9e7faaab83fe06e3761..38a161fc5132efabfc51044ec395865756841cbc 100644
--- a/dist/changesets-cli.cjs.js
+++ b/dist/changesets-cli.cjs.js
@@ -169,13 +169,6 @@ async function askQuestion(message) {
   });
 }
 
-function askQuestionWithEditor(message) {
-  const response = externalEditor.edit(message, {
-    postfix: ".md"
-  });
-  return response.replace(/^#.*\n?/gm, "").replace(/\n+$/g, "").trim();
-}
-
 async function askConfirm(message) {
   const name = `Confirm-${serialId()}`;
   return enquirer.prompt([{
@@ -405,35 +398,9 @@ async function createChangeset(changedPackages, allPackages) {
     });
   }
 
-  logger.log("Please enter a summary for this change (this will be in the changelogs).");
-  logger.log(chalk__default["default"].gray("  (submit empty line to open external editor)"));
-  let summary = await askQuestion("Summary");
-
-  if (summary.length === 0) {
-    try {
-      summary = askQuestionWithEditor("\n\n# Please enter a summary for your changes.\n# An empty message aborts the editor.");
-
-      if (summary.length > 0) {
-        return {
-          confirmed: true,
-          summary,
-          releases
-        };
-      }
-    } catch (err) {
-      logger.log("An error happened using external editor. Please type your summary here:");
-    }
-
-    summary = await askQuestion("");
-
-    while (summary.length === 0) {
-      summary = await askQuestion("\n\n# A summary is required for the changelog! 😪");
-    }
-  }
-
   return {
     confirmed: false,
-    summary,
+    summary: "",
     releases
   };
 }
